rimage/transform/parallel_projection.go

```diff
@@ -95,5 +97,162 @@ func (pp *ParallelProjection) PointCloudToRGBD(cloud pointcloud.PointCloud) (*ri
+// size. It will also add a red marker to the map to represent the location of the robot. The returned depthMap
+// is unused and so will always be nil.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) PointCloudToRGBD(cloud pointcloud.PointCloud,
+) (*rimage.Image, *rimage.DepthMap, error) {
+	meta := cloud.MetaData()
+
+	var err error
+	var X, Z []float64
+	cloud.Iterate(0, 0, func(pt r3.Vector, data pointcloud.Data) bool {
+		X = append(X, pt.X)
+		Z = append(Z, pt.Z)
+		return true
+	})
+
+	// Calculate max and min range to be represented in the output image and, if needed, cropping it based on
+	// the mean and standard deviation of the X and Z coordinates
+	meanX, err := stats.Mean(X)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "calculation of the X-coord's mean during pcd projection failed")
+	}
+	stdevX, err := stats.StandardDeviation(X)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "calculation of the X-coord's stdev during pcd projection failed")
+	}
+
+	meanZ, err := stats.Mean(Z)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "calculation of the Z-coord's mean during pcd projection failed")
+	}
+	stdevZ, err := stats.StandardDeviation(Z)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "calculation of the Z-coord's stdev during pcd projection failed")
+	}
+
+	maxX := math.Min(meanX+float64(sigmaLevel)*stdevX, meta.MaxX)
+	minX := math.Max(meanX-float64(sigmaLevel)*stdevX, meta.MinX)
+	maxZ := math.Min(meanZ+float64(sigmaLevel)*stdevZ, meta.MaxZ)
+	minZ := math.Max(meanZ-float64(sigmaLevel)*stdevZ, meta.MinZ)
+
+	// Change the max and min values to ensure the robot marker can be represented in the output image
+	var robotMarker spatialmath.Pose
+	if ppRM.robotPose != nil {
+		robotMarker = *ppRM.robotPose
+		maxX = math.Max(maxX, robotMarker.Point().X)
+		minX = math.Min(minX, robotMarker.Point().X)
+		maxZ = math.Max(maxZ, robotMarker.Point().Z)
+		minZ = math.Min(minZ, robotMarker.Point().Z)
+	}
+
+	// Calculate the scale factors
+	var scaleFactor float64
+	if (maxX-minX) != 0 || (maxZ-minZ) != 0 {
+		widthScaleFactor := float64(imageWidth-1) / (maxX - minX)
+		heightScaleFactor := float64(imageHeight-1) / (maxZ - minZ)
+		scaleFactor = math.Min(widthScaleFactor, heightScaleFactor)
+	}
+
+	// Add points in the pointcloud to a new image
+	var voxelColor rimage.Color
+	im := rimage.NewImage(imageWidth, imageHeight)
+	cloud.Iterate(0, 0, func(pt r3.Vector, data pointcloud.Data) bool {
+		x := int(math.Round((pt.X - minX) * scaleFactor))
+		y := int(math.Round((pt.Z - minZ) * scaleFactor))
+
+		// Adds a point to an image using the value to define the color. If no value is available,
+		// the default color of white is used.
+		if x >= 0 && x < imageWidth && y >= 0 && y < imageHeight && data != nil {
+			voxelColor, err = getProbabilityColorFromValue(data)
+			if err != nil {
+				return false
+			}
+			im.Circle(image.Point{x, y}, voxelRadius, voxelColor)
```

Comment:
I would expect this to just set the pixel, instead of drawing a circle.

Reply:
I imagine this might be used because setting a pixel might not be visible enough? Though even if that's the case, we probably want to use a different `voxelRadius` for the pointcloud data vs. the marker.

[q] You're using `im.Circle(image.Point{x, y}, voxelRadius, voxelColor)` and `im.Circle(image.Point{X: x, Y: y}, voxelRadius, robotMarkerColor)` below. Can you use the same style for both?

Reply:
Yeah, I think it's okay for this to either just set the pixel or use a smaller voxelRadius for pointcloud points than you use for the robot marker.